examples/08_join_aggregation.py

```diff
@@ -1,302 +1,311 @@
-)
-X_transformed = agg_target_movie.fit_transform(X, y)
-X_transformed.shape
-###############################################################################
-X_transformed.head()
-
-###############################################################################
-# Chaining everything together in a pipeline
-# ------------------------------------------
-#
-# To perform cross-validation and enable hyper-parameter tuning, we gather
-# all elements into a scikit-learn |Pipeline| by using |make_pipeline|,
-# and define a scikit-learn |HGBR|.
-from sklearn.ensemble import HistGradientBoostingRegressor
-from sklearn.pipeline import make_pipeline
+# See `here <https://skrub-data.org/stable/auto_examples/01_encodings.html#specializing-the-tablevectorizer-for-histgradientboosting>`_
+# for more details about these choices.
+from sklearn.preprocessing import OrdinalEncoder
 
-pipeline = make_pipeline(
-    table_vectorizer,
-    agg_target_user,
-    agg_target_movie,
-    HistGradientBoostingRegressor(learning_rate=0.1, max_depth=4, max_iter=40),
-)
+from skrub import MinHashEncoder, TableVectorizer
 
-pipeline
+vectorizer = TableVectorizer(
+    high_cardinality=MinHashEncoder(),  # encode ["item", "model"]
+    specific_transformers=[
+        (OrdinalEncoder(), ["make", "goods_code"]),
+    ],
+)
+products_transformed = vectorizer.fit_transform(products)
+TableReport(products_transformed)
 
-###############################################################################
-# Hyper-parameters tuning and cross validation
-# --------------------------------------------
+# %%
+# Our objective is now to aggregate this dataframe by basket ID, then merging it
+# on the basket dataframe, still on the basket ID.
+#
+# AggJoiner can help us achieve exactly this. We need to pass the product dataframe as
+# an auxiliary table argument to AggJoiner in ``__init__``. ``aux_key`` represent both
+# the columns used to groupby on, and the columns used to join on.
 #
-# We can finally create our hyper-parameter search space, and use a
-# |GridSearchCV|. We select the cross validation splitter to be
-# the |TimeSeriesSplit| to prevent leakage, since our data are timestamped
-# logs.
+# The basket dataframe is our main table, and we indicate the columns to join with
+# ``main_key``. Note that we pass the main table during ``fit``, and we discuss the
+# limitations of this design in the conclusion at the bottom of this notebook.
 #
-# Note that you need the name of the pipeline elements to assign them
-# hyper-parameters search.
+# The minimum is the most appropriate operation to aggregate encodings from
+# MinHashEncoder, for reasons that are out of the scope of this notebook.
 #
-# You can lookup the name of the pipeline elements by doing:
-list(pipeline.named_steps)
+# Notice that the ``basket_ID`` column is dropped from the joined table.
+from skrub import AggJoiner
+from skrub import _selectors as s
+
+minhash_cols_query = s.glob("item*") | s.glob("model*")
```

Comment:
Maybe we should use a search of columns by hand if people are not familiar with selectors

Reply:
that's true but on the other hand it will add a bunch of annoying / distracting boilerplate. maybe we can add a comment saying what it does. 

it will get nicer if the aggjoiner accepts a selector directly for the `cols=`; this will remove the need to do the s.select ourselves

Reply:
+1 for a clear comment